cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/store/AppMetadataStore.java

```diff
@@ -311,34 +299,34 @@ public void deleteApplications(String namespaceId)
   }
 
   // todo: do we need appId? may be use from appSpec?
-  public void updateAppSpec(String namespaceId, String appId, String versionId, ApplicationSpecification spec)
-  throws IOException {
-    LOG.trace("App spec to be updated: id: {}: spec: {}", appId, GSON.toJson(spec));
-    ApplicationMeta existing = getApplication(namespaceId, appId, versionId);
-    ApplicationMeta updated;
+  public void updateAppSpec(String namespaceId, String appId, String versionId,
```

Comment:
if it's a small change, would be better to refactor this to take the ApplicationId directly instead of the individual components

Reply:
Fixed